core/src/test/java/com/linecorp/armeria/client/encoding/DecodingClientTest.java

```diff
@@ -89,48 +89,21 @@ protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) {
     };
 
     @Test
-    void httpGzipDecodingTest() {
-        final BlockingWebClient client =
-                server.blockingWebClient(cb -> {
-                    cb.decorator(DecodingClient.newDecorator(
-                            com.linecorp.armeria.common.encoding.StreamDecoderFactory.gzip()));
-                });
-
-        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
-            final AggregatedHttpResponse response =
-                    client.execute(RequestHeaders.of(HttpMethod.GET, "/encoding-test"));
-            assertContentEncoding(captor.get().log(), response, "gzip");
-            assertThat(response.contentUtf8()).isEqualTo("some content to compress more content to compress");
-        }
-    }
-
-    @Test
-    void httpDeflateDecodingTest() {
-        final BlockingWebClient client = server.blockingWebClient(cb -> {
-            cb.decorator(DecodingClient.newDecorator(
-                    com.linecorp.armeria.common.encoding.StreamDecoderFactory.deflate()));
-        });
-
-        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
-            final AggregatedHttpResponse response =
-                    client.execute(RequestHeaders.of(HttpMethod.GET, "/encoding-test"));
-            assertContentEncoding(captor.get().log(), response, "deflate");
-            assertThat(response.contentUtf8()).isEqualTo("some content to compress more content to compress");
-        }
-    }
-
-    @Test
-    void httpBrotliDecodingTest() {
-        final BlockingWebClient client = server.blockingWebClient(cb -> {
-            cb.decorator(DecodingClient.newDecorator(
-                    com.linecorp.armeria.common.encoding.StreamDecoderFactory.brotli()));
-        });
-
-        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
-            final AggregatedHttpResponse response =
-                    client.execute(RequestHeaders.of(HttpMethod.GET, "/encoding-test"));
-            assertContentEncoding(captor.get().log(), response, "br");
-            assertThat(response.contentUtf8()).isEqualTo("some content to compress more content to compress");
+    void httpDecodingTest() {
+        for (com.linecorp.armeria.common.encoding.StreamDecoderFactory factory
+                : com.linecorp.armeria.common.encoding.StreamDecoderFactory.all()) {
```

Comment:
We could make this a `@ParameterizedTest`?

Reply:
I used for-loop since Brotli is optionally available. I was too lazy to create a `ArgumentsProvider`. 😅 Let me convert it to `@ParameterizedTest`